tests/testthat/test-loopy-logging.R

```diff
@@ -96,8 +101,136 @@ test_that("prediction error doesn't stop grid", {
+  expect_identical(ncol(res_fit$.notes[[1]]), 3L)
+  expect_true(all(vapply(res_fit$.notes[[1]], is.character, logical(1))))
+})
+
+test_that("capturing error correctly in notes", {
+  assign("prep.step_logging_helper", prep.step_logging_helper, envir = .GlobalEnv)
+  assign("bake.step_logging_helper", bake.step_logging_helper, envir = .GlobalEnv)
+  ames <- modeldata::ames[, c(72, 40:45)]
+
+  set.seed(1234)
+  folds <- rsample::vfold_cv(ames, 2)
+
+  rec_spec <- recipe(Sale_Price ~ ., ames) %>%
+    step_logging_helper(type = "error")
+  mod_spec <- parsnip::nearest_neighbor(
+    "regression",
+    "kknn",
+    dist_power = tune()
+  )
+
+  wf_spec <- workflow(rec_spec, mod_spec)
+
+  res_fit <- melodie_grid(
+    wf_spec,
+    folds,
+    grid = 2,
+    control = control_grid(allow_par = FALSE)
+  )
+
+  exp <- tibble::tibble(
+    location = "preprocessor 1/1",
+    type = "error",
+    note = "testing error"
+  )
+  expect_identical(res_fit$.notes[[1]], exp)
+  expect_identical(res_fit$.notes[[2]], exp)
+
+  expect_true(
+    all(vapply(res_fit$.metrics, NROW, integer(1)) == 0)
+  )
+  rm(list = "prep.step_logging_helper", envir = .GlobalEnv)
+  rm(list = "bake.step_logging_helper", envir = .GlobalEnv)
+})
+
+test_that("capturing warning correctly in notes", {
+  assign("prep.step_logging_helper", prep.step_logging_helper, envir = .GlobalEnv)
+  assign("bake.step_logging_helper", bake.step_logging_helper, envir = .GlobalEnv)
+  
+  ames <- modeldata::ames[, c(72, 40:45)]
+
+  set.seed(1234)
+  folds <- rsample::vfold_cv(ames, 2)
+
+  rec_spec <- recipe(Sale_Price ~ ., ames) %>%
+    step_logging_helper(type = "warning")
+  mod_spec <- parsnip::nearest_neighbor(
+    "regression",
+    "kknn",
+    dist_power = tune()
+  )
+
+  wf_spec <- workflow(rec_spec, mod_spec)
+  res_fit <- melodie_grid(
+    wf_spec,
+    folds,
+    grid = 2,
+    control = control_grid(allow_par = FALSE)
+  )
+
+  exp <- tibble::tibble(
+    location = "preprocessor 1/1",
+    type = "warning",
+    note = "testing warning"
+  )
+  expect_identical(res_fit$.notes[[1]], exp)
+  expect_identical(res_fit$.notes[[2]], exp)
+
+  expect_true(
+    all(vapply(res_fit$.metrics, NROW, integer(1)) > 0)
+  )
+  rm(list = "prep.step_logging_helper", envir = .GlobalEnv)
+  rm(list = "bake.step_logging_helper", envir = .GlobalEnv)
+})
+
+test_that("doesn't capturing message in notes", {
+  assign("prep.step_logging_helper", prep.step_logging_helper, envir = .GlobalEnv)
+  assign("bake.step_logging_helper", bake.step_logging_helper, envir = .GlobalEnv)
+  
+  ames <- modeldata::ames[, c(72, 40:45)]
+
+  set.seed(1234)
+  folds <- rsample::vfold_cv(ames, 2)
+
+  rec_spec <- recipe(Sale_Price ~ ., ames) %>%
+    step_logging_helper(type = "message")
+  mod_spec <- parsnip::nearest_neighbor(
+    "regression",
+    "kknn",
+    dist_power = tune()
+  )
+
+  wf_spec <- workflow(rec_spec, mod_spec)
+
+  expect_snapshot(
+    res_fit <- tune_grid(
```

Comment:
I think that this should be `melodie_grid()`. As is, I get 

```
── Failed tests ─────────────────────────────────────────────────────────────────────────────────────────────────────────────
Failure (test-loopy-logging.R:213:3): doesn't capturing message in notes
Snapshot of code has changed:
old[1:5] vs new[1:5]
  Code
-   res_fit <- tune_grid(wf_spec, folds, grid = 2, control = control_grid(
+   res_fit <- melodie_grid(wf_spec, folds, grid = 2, control = control_grid(
      allow_par = FALSE))
  Message
    testing message

* Run testthat::snapshot_accept('loopy-logging') to accept the change.
* Run testthat::snapshot_review('loopy-logging') to interactively review the change.
```

No errors with it (besides a snapshot update since the code changed)